packages/devtools_app/lib/src/screens/memory/panes/chart/controller/chart_connection.dart

```diff
@@ -38,28 +40,59 @@ class ChartConnection extends DisposableController
   Timer? _pollingTimer;
   bool _connected = false;
 
-  late final isDeviceAndroid =
-      serviceConnection.serviceManager.vm?.operatingSystem == 'android';
+  /// If true, the connection to application was lost.
+  bool _disconnected = false;
+
+  late final isDeviceAndroid = _isDevToolsCurrentlyConnected()
+      ? serviceConnection.serviceManager.vm?.operatingSystem == 'android'
+      : false;
+
+  /// True if DevTools is in connected mode and the connection to the app is alive.
+  bool _isDevToolsCurrentlyConnected() =>
+      !offlineDataController.showingOfflineData.value &&
+      serviceConnection.serviceManager.connectedState.value.connected &&
+      serviceConnection.serviceManager.connectedApp != null;
 
   Future<void> maybeConnect() async {
```

Comment:
is it possible that maybeConnect will get called more than once? We should be  calling cancelStreamSubscriptions before we add the auto dispose subscriptions in this method

Reply:
Yes, but connection is established only once. Replaced booleans with completers to make it clearer. Thanks.